src/cli/commands/test/mod.rs

```
@@ -92,6 +94,111 @@ impl From<(String, TestStatus)> for TestResult {
+///     %{ mock_call(func_to_mock, [mock_ret_value]) %}
+///     %{ mock_call(name_of_my_func, [42, “shortstring”, ids.myVarName, Uint256 {high: 42, low: 0x0}]) %}
+///     let x = mocked_func();
+///     assert 42 = x;
+///     return ();
+/// }";
+/// let res = extract_fname_mock_values(cairo_test).unwrap();
+/// assert_eq!(res["test_mock_call"].len(), 2);
+/// assert_eq!(res["test_mock_call"][0].fn_name, "func_to_mock");
+/// assert_eq!(res["test_mock_call"][0].fn_args, "[mock_ret_value]");
+/// ```
+pub fn extract_fname_mock_values(data: &str) -> Result<MockedFn, Error> {
+	let fun_regex: Regex = Regex::new(FUNC_RX).unwrap();
+	let mock_regex: Regex = Regex::new(MOCK_RX).unwrap();
+
+	let reg_caps = fun_regex.captures(data).unwrap();
+	let fn_name = reg_caps.name("func_test_name").unwrap().as_str();
+
+	let mut vec_mock = Vec::new();
+	let mut res = MockedFn::new();
+
+	for cap in mock_regex.captures_iter(data) {
+		vec_mock.push(MockEntry {
+			fn_name: cap.name("func_to_mock").unwrap().as_str(),
+			fn_args: cap.name("mock_value").unwrap().as_str(),
+		});
+	}
+	res.insert(fn_name, vec_mock);
+	Ok(res)
+}
+
+/// compute the list of all mocked functions for all functions
+/// in a cairo test file
+/// param path:&PathBuf the path to the cairo file
+/// Return a Result<()>
+fn list_test_mock_call(path: &PathBuf) -> Result<(), TestCommandError> {
+	let fun_regex: Regex = Regex::new(FUNC_RX).unwrap();
+	let data = fs::read_to_string(path)?;
+
+	//prepare sections to parse, corresponding to functions body
+	let mut pos: Vec<usize> = fun_regex.find_iter(data.as_str()).map(|x| x.start()).collect();
+	// append the last line to parse entire file
+	if !pos.is_empty() {
+		pos.push(data.len());
+	}
+
+	//extract fname and mock values
+	let mut i = 0;
+	while i != pos.len() - 1 {
+		match extract_fname_mock_values(&data[pos[i]..pos[i + 1]]) {
+			Ok(res) => {
+				println!("{res:?}");
+			},
+			_ => {
+				println!("FAILURE");
```

Return the error there rather than ignoring it